arch/arm/src/common/arm_initialize.c

```diff
@@ -163,6 +164,10 @@ void up_initialize(void)
   note_register();      /* Non-standard /dev/note */
 #endif
 
+#if defined(CONFIG_DRIVER_NOTECTL)
+  notectl_register();   /* Non-standard /dev/notectl */
```

Comment:
with the PR: https://github.com/apache/incubator-nuttx/pull/1731
we can move notectl_register to drivers/note/note_driver.c, to avoid touch the arch file every time the new transport or filter drivrer is added.

Reply:
@xiaoxiang781216 I have reverted the notectl driver initialization in the all arch files. 54a2a2e
I'll follow your changes completely after #1731 is merged.